include/RAJA/policy/openmp/forall.hpp

```diff
@@ -156,6 +270,73 @@ namespace internal
     }
   }
 
+  //
+  // omp for schedule(dynamic) nowait
+  //
+  template <typename Iterable, typename Func, int ChunkSize,
+    typename std::enable_if<(ChunkSize <= 0)>::type* = nullptr>
+  RAJA_INLINE void forall_impl_nowait(const ::RAJA::policy::omp::Dynamic<ChunkSize>&,
+                               Iterable&& iter,
+                               Func&& loop_body)
+  {
+    RAJA_EXTRACT_BED_IT(iter);
+    #pragma omp for schedule(dynamic) nowait
+    for (decltype(distance_it) i = 0; i < distance_it; ++i) {
+      loop_body(begin_it[i]);
+    }
+  }
+
+  //
+  // omp for schedule(dynamic, ChunkSize) nowait
+  //
+  template <typename Iterable, typename Func, int ChunkSize,
+    typename std::enable_if<(ChunkSize > 0)>::type* = nullptr>
+  RAJA_INLINE void forall_impl_nowait(const ::RAJA::policy::omp::Dynamic<ChunkSize>&,
+                               Iterable&& iter,
+                               Func&& loop_body)
+  {
+    RAJA_EXTRACT_BED_IT(iter);
+    #pragma omp for schedule(dynamic, ChunkSize) nowait
```

Comment:
Are you sure we want to provide the nowait versions of the dynamic policies?  They work, in that they will prevent a barrier, but unless you manually insert one or use some other sync mechanism there's no way to safely use the result in a subsequent loop.

Reply:
I thought about that, but wanted to know what folks thought.  I'll pull out the nowait versions for all non-static policies if you think that's best.

Reply:
> I personally do. It’s possible to use that correctly, but quite hard, and it would require writing non-idiomatic RAJA code. I admit I keep looking at this and wondering why OpenMP doesn’t have an option for conditional nowait though. There’s no reason that necessarily has to be a compile-time decision. Kinda makes me want to make them all nowait and have a barrier after the loop with an if-clause so we can collapse out the duplication. - @trws 

@trws I understand how it wouldn't necessarily work if you were to use the same data in the subsequent loop for a dynamic policy, but would it not be useful to have a nowait if the subsequent loop were performing work on different memory?